projects/ama-sdk/src/lib/services/editor-helper.service.ts

```diff
@@ -0,0 +1,41 @@
+ /*!
+ * @license
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { MatTabChangeEvent } from '@angular/material';
+import { ToolbarMessageAction } from '../store/app.actions';
+import { Store } from '@ngrx/store';
+import { AmaState } from '../store/app.state';
+import { CodeEditorPosition } from './../code-editor/components/code-editor/code-editor.component';
+
+@Injectable()
+export class EditorHelperService {
```

Comment:
selectedTabIndex is something which belongs to the actual editor. In case of process we have 3 tabs, in case of the others we have only 2. I'm afraid, we can't make it reusable this way. What we should do, is to create a common base for all the editors which we extend or compose. But for now, please let's move this back to components regardless of the introduced duplication.

Reply:
Should be provided in the root?

Reply:
I think this service should be deleted completely.